cromwell-backend/src/main/scala/cromwell/backend/Backend.scala

```diff
@@ -0,0 +1,39 @@
+package cromwell.backend
+
+import cromwell.backend.model.Subscription
+import cromwell.caching.Cacheable
+
+/**
+  * Defines basic functionality to interact with a Backend.
+  */
+trait Backend extends Cacheable {
+  /**
+    * Prepare the task and context for execution.
+    */
+  def prepare(): Unit
+
+  /**
+    * Executes task in given context.
+    */
+  def execute(): Unit
+
+  /**
+    * Stops a task execution.
+    */
+  def stop(): Unit
+
+  /**
+    * Performs a cleanUp after the task was executed.
+    */
+  def cleanUp(): Unit
+
+  /**
+    * Subscribe to events on backend.
+    */
+  def subscribeToEvent[T](subscription: Subscription[T]): Unit
+
+  /**
+    * Unsubscribe to events on backend.
+    */
+  def unsubscribeToEvent[T](subscription: Subscription[T]): Unit
+}
```

Comment:
@geoffjentry Can you provide us with the scalastyle_config.xml you are using?... In that way there will not be the need of taking care about code convention issues in the review.